make setup fails because uv installs to user-local PATH not available in subsequent make target

Running sudo make setup fails at the uv sync --dev step with make: uv: No such file or directory.
The issue is that the uv install script installs to $HOME/.local/bin (which resolves to /root/.local/bin under sudo), but that directory isn't in PATH when make proceeds to run uv sync.

sudo make setup
```

**Expected Behavior:**
Setup completes successfully, including `uv sync --dev`.

**Actual Behavior:**
System dependencies install fine, but the final step fails:
```
uv sync --dev
make: uv: No such file or directory
make: *** [Makefile:19: setup] Error 127

Suggested Fix:
After installing uv, source the env or reference uv by its full path before calling uv sync. For example in the Makefile:

export PATH := $(HOME)/.local/bin:$(PATH)

Or source the env inline:

setup:
	curl -LsSf https://astral.sh/uv/install.sh | sh
	. $(HOME)/.local/bin/env && uv sync --dev

Environment:
Ubuntu 24.04
uv 0.10.2
GNU Make

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -17,8 +17,9 @@ cd satellite
 make setup
 ```
 
-> **Linux note:** `make setup` will use `sudo` to install system libraries (libcairo2-dev).
-> Do **not** run `sudo make setup` — this will break the Python environment.
+> **Linux note:** `make setup` may use `sudo` to install system libraries (libcairo2-dev).
+> Running `sudo make setup` is supported: Python dependencies are installed as the invoking user (`$SUDO_USER`), not in `/root`.
+> Running `make setup` directly as root (without `SUDO_USER`) is not supported.
 > If you prefer to install system deps separately:
 > ```bash
 > sudo apt-get install -y libcairo2-dev pkg-config

--- a/Makefile
+++ b/Makefile
@@ -21,13 +21,22 @@ deps:
 
 .PHONY: setup
 setup: deps
-	@if [ "$$(id -u)" = "0" ]; then \
-		echo "Error: do not run 'make setup' as root or with sudo."; \
-		echo "Run 'make deps' for system packages, then 'make setup' as your normal user."; \
+	@if [ "$$(id -u)" = "0" ] && [ -n "$${SUDO_USER:-}" ]; then \
+		echo "Running Python setup as $$SUDO_USER..."; \
+		sudo -u "$$SUDO_USER" -H sh -lc 'command -v uv >/dev/null 2>&1 || { echo "Installing uv..."; curl -LsSf https://astral.sh/uv/install.sh | sh; }; UV_BIN="$$HOME/.local/bin/uv"; if [ -x "$$UV_BIN" ]; then "$$UV_BIN" sync --dev; else uv sync --dev; fi'; \
+	elif [ "$$(id -u)" = "0" ]; then \
+		echo "Error: running 'make setup' directly as root is not supported."; \
+		echo "Run 'make setup' as a normal user (or via sudo from that user account)."; \
 		exit 1; \
+	else \
+		command -v uv >/dev/null 2>&1 || { echo "Installing uv..."; curl -LsSf https://astral.sh/uv/install.sh | sh; }; \
+		UV_BIN="$$HOME/.local/bin/uv"; \
+		if [ -x "$$UV_BIN" ]; then \
+			"$$UV_BIN" sync --dev; \
+		else \
+			uv sync --dev; \
+		fi; \
 	fi
-	@command -v uv >/dev/null 2>&1 || { echo "Installing uv..."; curl -LsSf https://astral.sh/uv/install.sh | sh; }
-	@. "$(HOME)/.local/bin/env" 2>/dev/null || true; uv sync --dev
 
 .PHONY: test
 test:

--- a/README.md
+++ b/README.md
@@ -35,8 +35,9 @@ make setup
 
 `make setup` installs Python 3.13, all dependencies, and registers the `satellite` command.
 
-> **Linux note:** `make setup` will use `sudo` to install system libraries (libcairo2-dev).
-> Do **not** run `sudo make setup` — this will break the Python environment.
+> **Linux note:** `make setup` may use `sudo` to install system libraries (libcairo2-dev).
+> Running `sudo make setup` is supported: Python dependencies are installed as the invoking user (`$SUDO_USER`), not in `/root`.
+> Running `make setup` directly as root (without `SUDO_USER`) is not supported.
 > If you prefer to install system deps separately:
 > ```bash
 > sudo apt-get install -y libcairo2-dev pkg-config

--- a/tests/test_makefile_setup.py
+++ b/tests/test_makefile_setup.py
@@ -1,7 +1,7 @@
 """Tests for Makefile setup-target safety.
 
-Verifies that ``make setup`` rejects root, sources the uv PATH correctly,
-and keeps sudo-requiring commands in the separate ``deps`` target.
+Verifies that ``make setup`` handles sudo safely, resolves uv reliably,
+and keeps system package installation in the separate ``deps`` target.
 """
 
 import os
@@ -24,39 +24,38 @@ def _dry_run_target(target: str) -> subprocess.CompletedProcess[str]:
 
 
 # ============================================================================
-# Root guard
+# Sudo handling
 # ============================================================================
 
 
-class TestSetupRejectsRoot:
-    """``make setup`` must refuse to run as root."""
+class TestSetupSudoFlow:
+    """``make setup`` should support sudo while preventing direct-root usage."""
 
-    def test_setup_contains_uid_check(self) -> None:
+    def test_setup_runs_python_setup_as_sudo_user(self) -> None:
         result = _dry_run_target("setup")
-        assert "id -u" in result.stdout
+        assert 'sudo -u "$SUDO_USER" -H sh -lc' in result.stdout
 
-    def test_setup_exits_nonzero_for_root(self) -> None:
+    def test_setup_rejects_direct_root_without_sudo_user(self) -> None:
         result = _dry_run_target("setup")
+        assert "directly as root is not supported" in result.stdout
         assert "exit 1" in result.stdout
 
 
 # ============================================================================
-# uv PATH resolution
+# uv path resolution
 # ============================================================================
 
 
-class TestUvEnvSourced:
-    """After a fresh uv install the env must be sourced before ``uv sync``."""
+class TestUvPathResolution:
+    """After a fresh uv install, setup must still run ``uv sync --dev``."""
 
-    def test_setup_sources_local_bin_env(self) -> None:
+    def test_setup_references_local_uv_binary(self) -> None:
         result = _dry_run_target("setup")
-        assert ".local/bin/env" in result.stdout
+        assert ".local/bin/uv" in result.stdout
 
-    def test_uv_sync_on_same_line_as_env_source(self) -> None:
+    def test_setup_runs_uv_sync_dev(self) -> None:
         result = _dry_run_target("setup")
-        lines = result.stdout.strip().splitlines()
-        env_and_sync = [line for line in lines if ".local/bin/env" in line and "uv sync" in line]
-        assert len(env_and_sync) == 1
+        assert "sync --dev" in result.stdout
 
 
 # ============================================================================
@@ -71,12 +70,12 @@ class TestDepsTarget:
         result = _dry_run_target("deps")
         assert result.returncode == 0
 
-    def test_setup_does_not_call_sudo(self) -> None:
+    def test_setup_does_not_install_system_packages(self) -> None:
         deps_output = _dry_run_target("deps").stdout.strip()
         setup_output = _dry_run_target("setup").stdout.strip()
         setup_only = setup_output.replace(deps_output, "")
         non_echo = [line for line in setup_only.splitlines() if not line.lstrip().startswith("echo ")]
-        assert "sudo" not in "\n".join(non_echo)
+        assert "apt-get" not in "\n".join(non_echo)
 
 
 # ============================================================================

